### PR TITLE
Allow for more expressive comparisons

### PR DESCRIPTION
**Pull request by marcusmoore88**
I ended up pulling down another copy and making the additions instead of working on top of #446

### CODE REVIEW

File: src/Carbon/Carbon.php [hunk 1]
```diff
@@ -1327,6 +1412,18 @@ public function max(Carbon $dt = null)
     }
 
     /**
+     * Get the maximum instance between a given instance (default now) and the current instance.
+     *
+     * @param Carbon $dt
```
Comment (GrahamCampbell) [inline src/Carbon/Carbon.php]: missing |null

**Comment by marcusmoore88** [comment]
StyleCI complained a bunch. The code is there if you would like to pull it in.

**Comment by marcusmoore88** [comment]
Hey there it is! Code commit + 4 StyleCI fix commits...sorry about that.

**Comment by GrahamCampbell** [comment]
> StyleCI complained a bunch. 

You could have just downloaded an applied the diff StyleCI gave you the first time around you know. ;)

**Comment by GrahamCampbell** [comment]
https://styleci.io/analyses/X0xZjX/diff

File: src/Carbon/Carbon.php [hunk 2]
```diff
@@ -1313,6 +1385,18 @@ public function min(Carbon $dt = null)
     }
 
     /**
+     * Get the minimum instance between a given instance (default now) and the current instance.
+     *
+     * @param Carbon $dt
```
Comment (GrahamCampbell) [inline src/Carbon/Carbon.php]: Carbon|null

File: tests/Carbon/ExpressiveComparisonTest.php [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+use Carbon\Carbon;
+use Tests\AbstractTestCase;
+
+class ExpressiveComparisonTest extends AbstractTestCase
+{
+
```
Comment (GrahamCampbell) [inline tests/Carbon/ExpressiveComparisonTest.php]: please remove this extra lie

**Comment by marcusmoore88** [comment]
I did not know that. First time seeing StyleCI in action. 

Thanks for the heads up.

**Comment by briannesbitt** [comment]
Still waiting on 2 more fixes ?

**Comment by marcusmoore88** [comment]
I think it's good? It's showing all 3 checks have passed for me...

**Comment by briannesbitt** [comment]
2 inline comments still outstanding?

**Comment by marcusmoore88** [comment]
Sorry, looks ok to me?
screen shot 2015-11-12 at 2 31 15 pm

File: src/Carbon/Carbon.php [hunk 4]
```diff
@@ -1189,6 +1189,18 @@ public function eq(Carbon $dt)
     }
 
     /**
+     * Determines if the instance is equal to another
+     *
+     * @param Carbon $dt
+     *
```
Comment (lucasmichot) [inline src/Carbon/Carbon.php]: I would add `@see aliasedFunctionName` to all this newly introduced functions.

**Comment by GrahamCampbell** [comment]
> Sorry, looks ok to me?

Inline comments won't change the status. Please address those comments.

**Comment by marcusmoore88** [comment]
*cries to self*

Everything ok now?

File: src/Carbon/Carbon.php [hunk 5]
```diff
@@ -1189,6 +1189,20 @@ public function eq(Carbon $dt)
...
+     * @see eq
```
Comment (lucasmichot) [inline src/Carbon/Carbon.php]: eq()
Same for the other sees

**Comment by lucasmichot** [comment]
Can you squash your commits ? ;-)

**Comment by marcusmoore88** [comment]
Haha. Never squashed commits before. Tried to and screwed up my local branch.

If you would like to pull it in as is then go for it. If not that's fine too.

**Comment by marcusmoore88** [comment]
This is driving me to drink. Finally got the commits squashed and StyleCI fails again....

Let me give it another go.

**Pull request merged**